module/os/linux/zfs/zpl_inode.c

```diff
@@ -593,7 +593,8 @@ zpl_link(struct dentry *old_dentry, struct inode *dir, struct dentry *dentry)
 
 	crhold(cr);
 	ip->i_ctime = current_time(ip);
-	igrab(ip); /* Use ihold() if available */
+	/* Shouldn't be first ref, so igrab() cannot return NULL */
+	VERIFY3P(igrab(ip), !=, NULL); /* Use ihold() if available */
```

Comment:
```suggestion
	VERIFY3P(igrab(ip), !=, NULL);
```

The `ihold()` comment here is out of date and we should drop it.